src/ucp/wireup/select.c

```diff
@@ -906,32 +946,31 @@ static ucs_status_t ucp_wireup_add_am_lane(ucp_wireup_select_ctx_t *select_ctx)
     criteria.calc_score         = ucp_wireup_am_score_func;
     ucp_wireup_fill_ep_params_criteria(&criteria, select_ctx->params);
 
-    if (ucs_test_all_flags(ucp_ep_get_context_features(ep),
+    if (ucs_test_all_flags(ucp_ep_get_context_features(select_ctx->ep),
                            UCP_FEATURE_TAG | UCP_FEATURE_WAKEUP)) {
         criteria.local_iface_flags |= UCP_WORKER_UCT_UNSIG_EVENT_CAP_FLAGS;
     }
 
-    status = ucp_wireup_select_transport(ep, select_ctx->address_list,
-                                         select_ctx->address_count, &criteria,
-                                         -1, -1, -1, -1, 1, &select_ctx->am_info);
-    if (status != UCS_OK) {
-        return status;
-    }
-
-    am_info      = &select_ctx->am_info;
-    dst_md_index = select_ctx->address_list[am_info->addr_index].md_index;
-
-    /* If the remote side is not p2p and has only signaled-am wakeup, it may
-     * deactivate its interface and wait for signaled active message to wake up.
-     * Use a proxy lane which would send the first active message as signaled to
-     * make sure the remote interface will indeed wake up. */
-    remote_cap_flags = select_ctx->address_list[am_info->addr_index].iface_attr.cap_flags;
-    is_proxy         = ucp_wireup_is_lane_proxy(ep, am_info->rsc_index,
-                                                remote_cap_flags);
+    return ucp_wireup_select_transport(select_ctx, &criteria, -1, -1, -1, -1,
+                                       show_error, &select_ctx->am_info);
+}
 
-    ucp_wireup_add_lane_desc(select_ctx, am_info, dst_md_index,
-                             UCP_WIREUP_LANE_USAGE_AM, is_proxy);
+static ucs_status_t
+ucp_wireup_add_am_lane(ucp_wireup_select_ctx_t *select_ctx,
+                       ucs_status_t am_status)
+{
+    if (!ucp_wireup_is_am_required(select_ctx)) {
+        return UCS_OK;
+    } else if (am_status != UCS_OK) {
+        /* Try again with error printing */
```

Comment:
this comment seems weird, try what again?

Reply:
removed